### ISSUE TITLE
Accessibility - Heading levels should only increase by 1

### ISSUE THREAD

**Issue by metamoni** [joemasilotti/railsdevs.com#435]
**Describe the bug**
There are a few places in the app where `h3`s are used, but we're skipping one level as there's no `h2` on the page. Applying headings correctly makes it much easier for screen reader users to navigate the page and find what they're looking for, so we should be using headings in a logical order and style them as needed.

**To Reproduce**
1. Go to 'railsdevs.com'
2. Scan the page using the [axe Chrome extension](https://www.deque.com/axe/)
3. Notice the warning "Heading levels should only increase by one"
4. The same warning appears on https://railsdevs.com/developers and the developer profile edit page

**Expected behavior**
- the heading level on these pages should instead be`h2`
- the above warning should disappear when you scan each page with axe
- the styling shouldn't change

**Screenshots**
_Homepage_
![Screenshot 2022-05-21 at 14 01 55](https://user-images.githubusercontent.com/22390758/169652942-53894e43-d9d5-4d5c-8828-5ef6ecb799e4.png)

_Developers page_
![Screenshot 2022-05-21 at 14 34 40](https://user-images.githubusercontent.com/22390758/169654054-b1c7a0cd-384f-4d74-b2f6-55a607c902f8.png)

_Edit profile page_
![Screenshot 2022-05-21 at 14 35 04](https://user-images.githubusercontent.com/22390758/169654072-41fc9149-175c-4c40-b6e1-21c23fa4e38d.png)

**Additional context**
If you'd like to learn more about why this matters, this article on [Deque University](https://dequeuniversity.com/rules/axe/4.0/heading-order) is a good place to start.

**Comment by metamoni** [comment]
Already on it 🙂

**Comment by joemasilotti** [comment]
Great find, thanks for pointing this out! I didn't know about the axe Chrome extension, definitely going to use this more in the future.

I merged your PR, too.
